desktop/env.js

```diff
@@ -5,23 +5,35 @@
  */
 const path = require( 'path' );
 const app = require( 'electron' ).app;
-const fs = require( 'fs' );
-const dialog = require( 'electron' ).dialog;
+const { existsSync, mkdirSync } = require( 'fs' );
 
 /**
- * Internal dependencies
+ * Initialize core components
  */
+
+const state = require( './lib/state' );
 const config = require( './lib/config' );
+const appData = path.join( app.getPath( 'appData' ), config.appPathName );
+
+// Initialize log directory prior to requiring any modules that log
+const logPath = process.env.WP_DEBUG_LOG ? process.env.WP_DEBUG_LOG : path.join( appData, 'logs', 'wp-desktop.log' );
+if ( ! existsSync( path.dirname( logPath ) ) ) {
+	mkdirSync( path.dirname( logPath ), { recursive: true }, ( err ) => {
+		if ( err ) {
+			err.message = 'Failed to initialize log directory: ' + err.message;
+			throw err;
+		}
+	} );
+}
```

Comment:
there shouldn't be a need for `existsSync()` right? We _always_ want to make sure the directory is there and so adding the `existsSync()` only creates a race-condition where the directory could be there when it's called and gone by the time it's done, leaving us in the same state as if we just assumed it was there.

`mkdirSync()` shouldn't fail if the directory already exists.

```js
const logPath = process.env.WP_DEBUG_LOG || path.join( appData, 'logs', 'wp-desktop.log' );
mkdirSync( path.dirname( logPath ), { recursive: true }, error => { … } );
```

Reply:
Thanks - good catch. Addressed in d7d3aea.

Reply:
Interestingly enough, a plain`mkdirSync( ... { recursive: true} )` will throw an error (code `EEXIST`) if the directory already exists (at least, it does in CI). Updated with try/catch in 029cb26 and it's working again as expected.